internal/pkg/agent/application/application.go

```diff
@@ -155,6 +158,7 @@ func New(
 			compModifiers = append(compModifiers, FleetServerComponentModifier(cfg.Fleet.Server),
 				InjectFleetConfigComponentModifier(cfg.Fleet, agentInfo),
 				EndpointSignedComponentModifier(),
+				InjectAPMConfig,
```

Comment:
Is there ever a time that `InjectAPMConfig` is not used? If its always used we should probably just place it directly in the component model generator. That would ensure that it matches the same as `elastic-agent inspect`.

Reply:
The only case is the testing mode on top of the if/else block... Since that's something coming from an integration test I didn't want the agent to touch anything...

Reply:
I think even in testing you would want it to be the same as well. If a test provides the configuration for enabling APM, then it should be enabled in the test as well.